### ISSUE TITLE
Alternative method needed for PHP 8 as will remove argument evaluation/modification

### ISSUE THREAD

**Issue by mc12345678** [mc12345678/Stock_By_Attributes_Combined#95]
https://github.com/mc12345678/Stock_By_Attributes_Combined/blob/1c60f469f5dde1c2c19e8ada6995e96cea775fd1/includes/classes/observers/class.products_with_attributes_stock.php#L138

Will be affected by: 
https://github.com/php/php-src/blob/67f8e938fd03398b19f6e6fa012f65e74c8ec82a/UPGRADING#L216-L218
Which identifies:
````
  . debug_backtrace() and Exception::getTrace() will no longer provide
    references to arguments. It will not be possible to change function
    arguments through the backtrace.
````

**Comment by mc12345678** [comment]
Looks like perhaps the best way to address this is to "stuff" data into the call to the two associated functions that address stock, capture it in the notifier, evaluate, possibly set some internal variables to further pass information along between related functions and then provide the needed response.

Oh well, the above seemed pretty capable, but based on the early report of the upcoming version something different will be needed for future PHP versions. If they only eliminated the ability to edit the arguments then I wouldn't have an issue, but they are being completely removed...

**Comment by mc12345678** [comment]
Upon further review, it appears that generally speaking the provided `$products_id` includes the hashed attributes which should then be locatable either in the `$order` or the `$_SESSION['cart']` from which then the specific attributes can be reviewed/retrieved.  There is one instance in default Zen Cart code where it appears that specifically the base product itself is expected to be reviewed which is something that is also supported by SBA when addressing available product.

Any ways, similar usage/review needs to be performed by code generated for this module to ensure that a similar lookup can be performed/is fully supported so that *any* call to the two functions in question can support the quantity lookup that is needed. This will involve a little more coding to replace the existing work around that was generated, but at least a solution appears viable to continue support without adjusting/touching internal base Zen Cart code.

The "hard" part will be identifying whether the information is to be sought from the `$order` or the `$_SESSION['cart']` especially in situations where both may exist simultaneously but mean different things...

**Comment by mc12345678** [comment]
Further investigation has indicated that the issue "corrected" by changes to PHP 7.4 and 8.0+ do not relate to how this code is used, but instead in how PHP had the potential to be abused... At any rate there is now a default php.ini setting that prevents the sharing of function/method parameter presentation. It can be overridden to allow collection of that information, but is defaulted to not support it.

The "easy" solution is to permit the override: `zend.exception_ignore_args` to be set to `'false'` so that the arguments are returned.  Why is this important? Well, there are some functions within the code that specifically send attributes and other collected information to the sba observer. I guess it would be possible to combine files/information so that nothing has to be passed per se, it would just be present as a part of the observer; however, the current method looks backwards on what was sent and tries to evaluate the information that was sent.
